src/pages/HackathonSelection.jsx

```diff
@@ -0,0 +1,410 @@
+  width: min(100%, 300px);
+  height: auto;
+  animation: ${float} 3s ease-in-out infinite;
+`
+
+const ButtonContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  gap: 0.25rem;
+  z-index: 2;
+`
+
+const Button = styled.button`
+  position: relative;
+  width: 250px;
+  padding: 12px 24px;
+  z-index: 2;
+
+  font-size: 1.25rem;
+  font-weight: ${p => p.theme.typography.h1.weight};
+  font-family: ${p => p.theme.typography.headerFont};
+
+  background: ${props => props.color};
+  color: ${props => props.labelColor};
+  border: none;
+  border-radius: 8px;
+
+  cursor: pointer;
+  transition: background 0.3s ease;
+
+  &:hover {
+    background: ${props => props.hoverColor || props.color};
+  }
+
+  ${p => p.theme.mediaQueries.tabletLarge} {
+    width: ${props => (props.isUpNext ? '250px' : '175px')};
+    padding: 8px 0;
+    font-weight: 600;
+    font-size: 1.15rem;
+  }
+
+  ${p => p.theme.mediaQueries.mobile} {
+    font-size: 1.15rem;
+    width: ${props => (props.isUpNext ? '175px' : '125px')};
+  }
+`
+
+const ApplicationStatusTextDesktop = styled(H1)`
+  font-size: 1.15rem;
+  font-weight: 600;
+
+  ${p => p.theme.mediaQueries.tabletLarge} {
+    display: ${props => (props.isUpNext ? 'block' : 'none')};
+  }
+
+  ${p => p.theme.mediaQueries.mobile} {
+    font-size: 0.8rem;
+  }
+`
+
+const ApplicationStatusTextMobile = styled(H1)`
+  font-weight: 600;
+  display: none;
+
+  ${p => p.theme.mediaQueries.tabletLarge} {
+    display: ${props => (props.isUpNext ? 'none' : 'block')};
+    font-size: 1rem;
+  }
+
+  ${p => p.theme.mediaQueries.mobile} {
+    font-size: 0.8rem;
+  }
+`
+
+// helps group items in mobile view
+const MobileContainer = styled.div`
+  display: contents;
+
+  ${p => p.theme.mediaQueries.tabletLarge} {
+    display: ${props => (props.isUpNext ? 'contents' : 'flex')};
+    flex-direction: ${props => (props.isUpNext ? 'column' : 'row')};
+    gap: ${props => (props.isUpNext ? '0' : '10vw')};
+    justify-content: center;
+    align-items: center;
+  }
+`
+
+// example deadline format: February 22nd, 2025 at 11:59 PM (Pacific Time)
+const getApplicationStatusText = (applicationOpen, applicationDeadline) => {
+  const deadlineDate = moment(applicationDeadline, 'MMMM Do, YYYY [at] h:mm A')
+  const isDeadlinePassed = moment().isAfter(deadlineDate)
+
+  if (isDeadlinePassed) return 'Applications closed'
+  if (!applicationOpen) return 'Applications opening soon!'
+  return `Applications close on ${deadlineDate.format('MMM D, YYYY')}!`
+}
+
+const handleNavigation = (applicationOpen, visitWebsite, hackathonName, navigate) => {
+  if (applicationOpen || !visitWebsite) return navigate(`/app/${hackathonName.toLowerCase()}`)
+
+  const websites = {
+    'HackCamp': 'https://hackcamp.nwplus.io',
+    'nwHacks': 'https://nwhacks.io',
+    'cmd-f': 'https://cmd-f.nwplus.io',
+  }
+  window.open(websites[hackathonName], '_blank', 'noopener,noreferrer')
+}
+
+const HackathonCard = ({
```

Comment:
would it be better to move this to another file under components along with the related css / styling code above just so this file isn't too cluttered

Reply:
💯 🫡 ✅